### ISSUE TITLE
Editor lag when searching for a common term with hidden search pane

### ISSUE THREAD

**Issue by marcoms** [atom/find-and-replace#861]
### Prerequisites

* [-] Put an X between the brackets on this line if you have done all of the following:
    * [x] Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * [x] Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * [x] Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

When searching for a very common occurrence, such as `' '`, the editor will lag because of the large amount of matches. This is at least understandable, but the lag continues to manifest even when the pane is closed and search results are hidden (ESC).

### Steps to Reproduce

1. Search for a very common term
2. Close find-and-replace pane
3. Try scrolling

**Expected behavior:** Scrolling should be smooth due to search results not being rendered

**Actual behavior:** Scrolling is not smooth

**Reproduces how often:** 100%

### Versions

```fish
> atom-beta --version
Atom    : 1.15.0-beta1
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0
```

```fish
> apm-beta --verison
apm  1.15.3
npm  3.10.5
node 4.4.5 x64
python 2.7.13
git 2.11.1
```

find-and-replace version 0.206.1

**Comment by rsese** [comment]
@marcoms can you share an example file that will reproduce the behavior?

**Comment by marcoms** [comment]
E.g. see https://paste.sh/yXXEII_6#wNXxAbDjlAFqgPWBmMrjjQS2 in the atom editor.

Searching for ' ' in this file can cause scrolling lag, and after hitting ESC and closing the search the lag continues. To fix it, I have to remove the search then hit ESC

**Comment by rsese** [comment]
Thanks for sharing that - I just gave it a try and didn't notice any lag with your steps to reproduce.  Can you confirm what operating system you're running?  I'm on OX X 10.10.5.

**Comment by marcoms** [comment]
I'm on 10.12.3

If you can't reproduce with that sample, if you could try just to find a case where the number of occurrences produces lag, then after closing the search view the lag should persist, while with no search term entered, scrolling is fine.
I suppose its difficult to find a sample with consistent results because it would depend on the machine's performance which varies from user to user.

It's not such a critical bug, and its easy to work around. I just reported because I thought it was an error that the search results were still being processed even after the search view was closed.

**Comment by rsese** [comment]
> If you can't reproduce with that sample, if you could try just to find a case where the number of occurrences produces lag, then after closing the search view the lag should persist, while with no search term entered, scrolling is fine.
> I suppose its difficult to find a sample with consistent results because it would depend on the machine's performance which varies from user to user.

Yep good point :+1:  From your sample content, I created a 7MB file and tested with that - I definitely experienced a bit of general slowness (I think there are issues open in atom/atom about this general behavior with large files) but when I searched for a ` `, things froze up for a bit and I wasn't able to `Esc` out of search.  When I was able to `Esc` out of search however, I didn't have any trouble scrolling the large document.

I'm on 1.15.0-beta3 and find-and-replace 0.206.3.

> It's not such a critical bug, and its easy to work around. I just reported because I thought it was an error that the search results were still being processed even after the search view was closed.

Definitely thanks for reporting :+1:

I also think it seems like a bug, but since I can't reproduce, I'll go ahead and tag it as `needs-reproduction` for now in case someone else can give it a try.
